airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/components/id_incremental_cursor.py

```diff
@@ -0,0 +1,112 @@
+    """
+    Custom Incremental Cursor implementation to provide the ability to pull data using `id`(int) as cursor.
+    More info: https://developer.zendesk.com/api-reference/live-chat/chat-api/agents/#parameters
+
+    Attributes:
+        config (Config): connection config
+        field_name (Union[InterpolatedString, str]): the name of the field which will hold the cursor value for outbound API call
+        cursor_field (Union[InterpolatedString, str]): record's cursor field
+    """
+
+    config: Config
+    cursor_field: Union[InterpolatedString, str]
+    field_name: Union[InterpolatedString, str]
+    parameters: InitVar[Mapping[str, Any]]
+    _cursor: Optional[str] = field(repr=False, default=None)
+    message_repository: Optional[MessageRepository] = None
+
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        self._state: Optional[int] = None
+        self._interpolation = JinjaInterpolation()
+        self.cursor_field = InterpolatedString.create(self.cursor_field, parameters=parameters)
+        self.field_name = InterpolatedString.create(self.field_name, parameters=parameters)
+
+    def get_stream_state(self) -> StreamState:
+        return {self.cursor_field.eval(self.config): self._cursor} if self._cursor else {}
+
+    def set_initial_state(self, stream_state: StreamState) -> None:
+        """
+        Cursors are not initialized with their state. As state is needed in order to function properly, this method should be called
+        before calling anything else
+
+        :param stream_state: The state of the stream as returned by get_stream_state
+        """
+
+        self._cursor = stream_state.get(self.cursor_field.eval(self.config)) if stream_state else None
+        self._state = self._cursor if self._cursor else self._state
+
+    def close_slice(self, stream_slice: StreamSlice, most_recent_record: Optional[Record]) -> None:
+        last_record_cursor_value = most_recent_record.get(self.cursor_field.eval(self.config)) if most_recent_record else None
+        self._cursor = last_record_cursor_value if last_record_cursor_value else None
+
+    def stream_slices(self) -> Iterable[StreamSlice]:
+        """
+        Use a single Slice.
+        """
+        slice = StreamSlice(partition={}, cursor_slice={})
+        return [slice]
+
+    def get_request_params(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        return self._get_request_options(RequestOptionType.request_parameter, stream_slice)
+
+    def _get_request_options(self, option_type: RequestOptionType, stream_slice: StreamSlice):
+        options = {}
+        if self._state:
+            options[self.field_name.eval(self.config)] = self._state
+        return options
+
+    def should_be_synced(self, record: Record) -> bool:
```

Comment:
Where/when is this called? Semi-incremental is not yet part of the CDK and hence, I don't think this does anything

Reply:
This is the `abstractmethod` set in the super-class: 
https://github.com/airbytehq/airbyte/blob/0113c525548080f128258042d8663a415897ef8c/airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/cursor.py#L55

It has to be defined, even if it does nothing.

Reply:
I've covered this method with `unit_test`, so it's testable.